components/gitpod-protocol/src/util/analytics.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import Analytics = require("analytics-node");
+import { log } from './logging';
+
+export const IAnalyticsWriter = Symbol("IAnalyticsWriter");
+
+type Identity =
+    | { userId: string | number }
+    | { userId?: string | number; anonymousId: string | number };
+
+interface Message {
+    messageId?: string;
+}
+
+export type IdentifyMessage = Message & Identity & {
+    traits?: any;
+    timestamp?: Date;
+    context?: any;
+};
+
+export type TrackMessage = Message & Identity & {
+    event: string;
+    properties?: any;
+    timestamp?: Date;
+    context?: any;
```

Comment:
what is the context for?

Reply:
see https://segment.com/docs/connections/spec/common/ (search for context)